src/main/scala/com/github/tarao/namedcap/MultiMap.scala

```diff
@@ -19,14 +19,17 @@ class MultiMap(m: Map[String, Seq[String]])
     extends ImmutableMap[String, String] {
   def get(key: String): Option[String] = m.get(key).flatMap(_.headOption)
   def getAll(key: String): Seq[String] = m.getOrElse(key, Seq.empty)
-  def +[V >: String](kv: (String, V)): MultiMap = {
-    val (key, value) = kv
-    val list = m.getOrElse(key, Vector.empty) :+ value.asInstanceOf[String]
-    new MultiMap(m + (key -> list))
-  }
-  def -(key: String): MultiMap = new MultiMap(m - key)
   def iterator: Iterator[(String, String)] =
     m.iterator.flatMap(kv => kv._2.headOption.map(kv._1 -> _))
+  def removed(key: String): MultiMap = new MultiMap(m - key)
+  def updated[V1 >: String](key: String, value: V1): MultiMap = {
+    val list = m.getOrElse(key, Seq.empty) :+ value.asInstanceOf[String]
+    new MultiMap(m + (key -> list))
+  }
+  override def +[V >: String](kv: (String, V)): MultiMap = {
```

Comment:
I explicitly override `+`  to keep  a return type `MultiMap`. It would be better to override `-`, too,  but it is finalized.